Geometry.cs

```diff
@@ -130,6 +140,10 @@ class Polygon {
 class Drawing {
    public void Add (Polygon poly) {
       mPolys.Add (poly);
+      if (mConvexHull.Count > 0) { 
+         mConvexHull.AddRange (poly.Pts); 

```

Comment:
Note 1: This is dangerous - you are adding the new points into mConvexHull, and then calling ComputingConvexHull passing the same array. What if the implementation of ComputeConvexHull is updated in the future to use mConvexHull itself as the work buffer. 
This problem is called variable aliasing - you are using the same mConvexHull for the input (IEnumerable<Point2>) as the routine is computing as its output. Right now it works only because the implementation builds up a convex hull in a temporary buffer before assigning it to mConvexHull. 
As an alternative (since you only need an IEnumerable), you could pass mConvexHull.Concat(poly.Pts) to it.

Reply:
Changed to use 'mConvexHull.Concat(poly.Pts)'